frame/contracts/src/wasm/runtime.rs

```diff
@@ -801,6 +806,34 @@ define_env!(Env, <E: Ext>,
 		ctx.set_storage(key_ptr, value_ptr, value_len).map(|_| ())
 	},
 
+	// Replace contract code.
```

Comment:
I think this kinda undersells what's happening here. I'd add some more details here, and implications (e.g, that whole discussion you and Alex had about deterministic addressed)

Reply:
It has some description under the `# NOTE` section. I'll do some updates.